packages/components/src/components/as-histogram-widget/utils/draw.service.ts

```diff
@@ -293,9 +296,31 @@ function _delayFn(_d, i) {
   return i;
 }
 
-export function conditionalFormatter(value) {
-  if (value > 0 && value < 1) {
-    return decimalFormatter(value);
+function getFloatPrecision(value) {
+  const expValue = value.toString();
+  const expPos = expValue.indexOf('.');
+  return expPos > -1 ? expValue.slice(expPos + 1).length : 0;
+}
+
+function getDomainPrecision(domain) {
+  let domainPrecision = 0;
+  if (!(domain[0] instanceof Date)) {
+    const domainDiff = domain[domain.length - 1] as any - domain[0];
```

Comment:
This `any` is really a number, right?

I guess we shouldn't have mixed domains, and if we checked just above that the first element is not a Date, this should be a number.

Reply:
Yes, you're right 😅